Add dynamic form field

holtzmak opened the pull request:
Related to #134 .

In this PR I added two form utility classes:
1. `StringFormField`, meant to be used in other `FormFields`
2. `DynamicStringFormField`, meant to be used in other `FormFields`. A dynamic form editor.

Demo:


https://user-images.githubusercontent.com/32527219/108616373-76e55700-73d2-11eb-813b-26c14fea1f3d.mov


Unfortunately, the dynamic form rebuilds a lot. This is because the text editing controller in each field has an initial text, that, if the widget rebuilds causes it to have the initial text again. The widgets rebuild anytime there is a change, so I've elected to accept a lot of rebuilding for a robust form that saves without being told to.

holtzmak committed:
Run flutter pub upgrade

holtzmak committed:
Add beginnings of reusable, dynamic form fields

holtzmak committed:
Add StringFormField and test

holtzmak committed:
Add MultiStringFormField in preparation for other multi formfields

holtzmak committed:
Split multi string form field and string form field

holtzmak committed:
Regen CocoaPods after firestore plugin

holtzmak commented on app/lib/ui/views/active/form_field/multi_string_form_field.dart:
```diff
@@ -0,0 +1,80 @@
+  List<String> getList() => _fields.map((field) => field.getString()).toList();
+
+  void _deleteField(String field) {
+    setState(() {
```
- [x] Oops, this doesn't work the way it should. I should key all the list items.
https://stackoverflow.com/a/59390995

holtzmak replied:
Will add video after I fix this.

MummenRider commented on app/ios/Podfile.lock:
```diff
@@ -1,4 +1,227 @@
 PODS:
+  - abseil/algorithm (0.20200225.0):
```
Do you what are these pods for?  I've seen this before as part of my experimentation, but no idea how I got it

holtzmak replied:
Yeah, `Abseil` is a C++ library (funny name). https://cocoapods.org/pods/abseil 
 It's definitely required by FirebaseFirestore. https://cocoapods.org/pods/FirebaseFirestore (dependencies on the right side of the page)

MummenRider reviewed (approved):
LGTM!

holtzmak committed:
Add UUID dependency

holtzmak committed:
Updated the string and dynamic string form fields to actually work...

Needed object keys to uniquely identify widgets. Had to refactor how we handle
multiple fields in a dynamic form. This causes more widget rebuilds
but other methods of smartly holding widgets doesn't work well with form saves
because the initial test would come back instead of edited text.

holtzmak commented:
@MummenRider ready for re-review (minor review, see video I added in original comment).
